File: command/command/document/IDocumentItemFactory.cs
using command.storage;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace command.document
{
	public interface IDocumentItemFactory
	{
		IImage CreateImage(string path, uint width, uint height);
		IParagraph CreateParagraph(string text);
	}
}

File: command_java/Document/src/document/documentItem/ItemsCollection/IDocumentItemFactory.java
package document.documentItem.ItemsCollection;

import document.documentItem.image.IImage;
import document.documentItem.image.Size;
import document.documentItem.paragraph.IParagraph;
import storage.IResource;

import java.io.IOException;

public interface IDocumentItemFactory {

    IImage CreateImage(String path, Size size) throws IOException;

    IParagraph CreateParagraph(String text);
}
